docs/ui/components/Header/Header.tsx

```diff
@@ -0,0 +1,136 @@
+import { css } from '@emotion/react';
+import { theme, breakpoints, SearchIcon, iconSize } from '@expo/styleguide';
+import React, { useState } from 'react';
+
+import { Logo } from './Logo';
+import { Search } from './Search';
+import { ThemeSelector } from './ThemeSelector';
+
+import { Button } from '~/ui/components/Button';
+import { BOLD } from '~/ui/components/Text';
+import { MenuIcon } from '~/ui/foundations/icons';
+
+export const Header = () => {
+  const [isMobileSearchVisible, setMobileSearchVisible] = useState(false);
+  const [isMobileMenuVisible, setMobileMenuVisible] = useState(false);
+  return (
+    <>
+      <nav css={containerStyle}>
+        <div css={[columnStyle, leftColumnStyle]}>
+          <Logo />
+        </div>
+        <Search version="latest" css={hideOnMobileStyle} />
+        <div css={[columnStyle, rightColumnStyle, hideOnMobileStyle]}>
+          <ThemeSelector />
+        </div>
+        <div css={[columnStyle, rightColumnStyle, showOnMobileStyle]}>
+          <Button
+            theme="transparent"
+            css={[mobileButtonStyle, isMobileSearchVisible && mobileButtonActiveStyle]}
+            onClick={() => {
+              setMobileMenuVisible(false);
+              setMobileSearchVisible(prevState => !prevState);
+            }}>
+            <SearchIcon size={iconSize.small} color={theme.icon.default} />
+          </Button>
+          <Button
+            theme="transparent"
+            css={[mobileButtonStyle, isMobileMenuVisible && mobileButtonActiveStyle]}
+            onClick={() => {
+              setMobileSearchVisible(false);
+              setMobileMenuVisible(prevState => !prevState);
+            }}>
+            <MenuIcon />
+          </Button>
+        </div>
+      </nav>
+      {isMobileMenuVisible ? (
+        <nav css={[containerStyle, showOnMobileStyle]}>
+          <div css={[columnStyle, leftColumnStyle]}>
+            <BOLD>Theme</BOLD>
+          </div>
+          <div css={[columnStyle, rightColumnStyle]}>
+            <ThemeSelector />
+          </div>
+        </nav>
+      ) : null}
+      {isMobileSearchVisible ? (
```

Comment:
Since this is a boolean, you can also use something like:

```
{isMobileSearchVisible && (
  ...
)}
```

But that's like a personal preference, not mandatory 😄

Reply:
I was using this notation for a bit, but I have decided to switch back to full ternary:
* https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

Reply:
Since using `&&` do not pose any potential issues, I have decided to apply the suggestion! 🙂